Fabric.Terminology.SqlServer/Persistence/SqlClientTermValueSetRepository.cs

```diff
@@ -9,135 +9,163 @@
     using Fabric.Terminology.Domain;
     using Fabric.Terminology.Domain.Exceptions;
     using Fabric.Terminology.Domain.Models;
+    using Fabric.Terminology.Domain.Persistence;
+    using Fabric.Terminology.SqlServer.Caching;
     using Fabric.Terminology.SqlServer.Models.Dto;
-    using Fabric.Terminology.SqlServer.Persistence.DataContext;
     using Fabric.Terminology.SqlServer.Persistence.Factories;
-
-    using Microsoft.EntityFrameworkCore;
+    using Fabric.Terminology.SqlServer.Persistence.UnitOfWork;
 
     using Serilog;
 
-    internal class SqlClientTermValueSetRepository : IClientTermValueSetRepository
-    {
-        private readonly Lazy<ClientTermContext> clientTermContext;
+    using static Fabric.Terminology.Domain.Persistence.OperationHelper;
 
+    internal partial class SqlClientTermValueSetRepository : IClientTermValueSetRepository
+    {
         private readonly ILogger logger;
 
-        public SqlClientTermValueSetRepository(Lazy<ClientTermContext> clientTermContext, ILogger logger)
+        private readonly IClientTermValueUnitOfWorkManager uowManager;
+
+        private readonly IClientTermCacheManager cacheManager;
+
+        public SqlClientTermValueSetRepository(
+            ILogger logger,
+            IClientTermValueUnitOfWorkManager uowManager,
+            IClientTermCacheManager cacheManager)
         {
-            this.clientTermContext = clientTermContext;
             this.logger = logger;
+            this.uowManager = uowManager;
+            this.cacheManager = cacheManager;
         }
 
-        protected ClientTermContext ClientTermContext => this.clientTermContext.Value;
-
         public Maybe<IValueSet> GetValueSet(Guid valueSetGuid)
         {
-            var desc = this.ClientTermContext.ValueSetDescriptions.AsNoTracking()
-                .SingleOrDefault(x => x.ValueSetGUID == valueSetGuid);
+            return this.uowManager.GetValueSetDescriptionDto(valueSetGuid)
+                .Select(
+                    dto =>
+                        {
+                            var factory = new ValueSetBackingItemFactory();
+                            var item = factory.Build(dto);
+                            ((ValueSetBackingItem)item).IsCustom = true;
```

Comment:
This might be a code smell. It might be fine, but consider whether it really makes sense that we have to make this cast.